File: TwoPointer/128.longest-consecutive-sequence.py
from typing import List
from collections import Counter


# @leet start
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        ret, freq = 0, Counter(nums)

        for x in freq:
            if x - 1 in freq:
                continue
            y = x + 1
            while y in freq:
                y += 1
            ret = max(ret, y - x)

        return ret


# @leet end

File: TwoPointer/128.longest-consecutive-sequence.go
package leetcode

// @leet start
func longestConsecutive(nums []int) int {
	freq := map[int]int{}
	for _, v := range nums {
		freq[v]++
	}
	ret := 0
	for x := range freq {
		if freq[x-1] > 0 {
			continue
		}
		y := x + 1
		for freq[y] > 0 {
			y++
		}
		ret = max(ret, y-x)
	}
	return ret
}

// @leet end

